File: TeamCode/src/main/java/org/manchestermachinemakers/hardware/drivebase/config/ProgrammingBoard.java
package org.manchestermachinemakers.hardware.drivebase.config;

import com.qualcomm.robotcore.hardware.DcMotorSimple;

import static org.manchestermachinemakers.hardware.drivebase.DriveBase.TravelDirection;

import org.manchestermachinemakers.hardware.drivebase.DriveBase;

import java.util.HashMap;

public class ProgrammingBoard extends DriveBase.Configuration {
    @Override
    public void init() {
        setMotors("left_front");
        put(TravelDirection.forward, DcMotorSimple.Direction.FORWARD);
        put(TravelDirection.reverse, DcMotorSimple.Direction.REVERSE);
        put(TravelDirection.pivotLeft, DcMotorSimple.Direction.REVERSE);
        put(TravelDirection.pivotRight, DcMotorSimple.Direction.FORWARD);
        put(TravelDirection.strafeLeft, DcMotorSimple.Direction.FORWARD);
        put(TravelDirection.strafeLeftBackward, DcMotorSimple.Direction.REVERSE);
        put(TravelDirection.strafeLeftForward, DcMotorSimple.Direction.FORWARD);
        put(TravelDirection.strafeRight, DcMotorSimple.Direction.REVERSE);
        put(TravelDirection.strafeRightBackward, DcMotorSimple.Direction.FORWARD);
        put(TravelDirection.strafeRightForward, DcMotorSimple.Direction.REVERSE);
        put(TravelDirection.pitch, DcMotorSimple.Direction.FORWARD);
    }
}

File: TeamCode/src/main/java/org/manchestermachinemakers/hardware/drivebase/config/ProgrammingBoard.kt
package org.manchestermachinemakers.hardware.drivebase.config

import org.manchestermachinemakers.hardware.drivebase.DriveBase.TravelDirection
import com.qualcomm.robotcore.hardware.DcMotorSimple
import org.manchestermachinemakers.hardware.drivebase.DriveBase

val programmingBoard = DriveBase.Configuration(
        arrayOf("left_front"),
        hashMapOf(
            TravelDirection.forward to arrayOf(DcMotorSimple.Direction.FORWARD),
            TravelDirection.reverse to arrayOf(DcMotorSimple.Direction.REVERSE),
            TravelDirection.pivotLeft to arrayOf(DcMotorSimple.Direction.REVERSE),
            TravelDirection.pivotRight to arrayOf(DcMotorSimple.Direction.FORWARD),
            TravelDirection.strafeLeft to arrayOf(DcMotorSimple.Direction.FORWARD),
            TravelDirection.strafeLeftBackward to arrayOf(DcMotorSimple.Direction.REVERSE),
            TravelDirection.strafeLeftForward to arrayOf(DcMotorSimple.Direction.FORWARD),
            TravelDirection.strafeRight to arrayOf(DcMotorSimple.Direction.REVERSE),
            TravelDirection.strafeRightBackward to arrayOf(DcMotorSimple.Direction.FORWARD),
            TravelDirection.strafeRightForward to arrayOf(DcMotorSimple.Direction.REVERSE),
            TravelDirection.pitch to arrayOf(DcMotorSimple.Direction.FORWARD),
        )
)